src/js/providers/tracks-mixin.js

```diff
@@ -102,7 +102,11 @@ define(['../utils/underscore',
         if (this._renderNatively) {
             // Only bind and set this.textTrackChangeHandler once so that removeEventListener works
             this.textTrackChangeHandler = this.textTrackChangeHandler || textTrackChangeHandler.bind(this);
-            this.addTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
+
+            this.removeTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
+            if (!this.instreamMode) {
+                this.addTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
+            }
 
             if (utils.isEdge()) {
```

Comment:
@hghazzi, yeah, it does. We should always remove existing listeners before adding one; there are side effects if we don't. Until we have to worry about displaying captions during ads, we don't need to listen to track changes while playing an ad. Also, one of my [PRs](https://github.com/jwplayer/jwplayer/pull/1374/files#diff-09efbcfa88512d3674c2d08e2b8a9496R275) that got merged yesterday made some updates to removing and adding listeners. You'll need to rebase to get those changes.